cocos/core/data/class.ts

```diff
@@ -455,8 +456,11 @@ function parseAttributes (constructor: Function, attributes: PropertyStash, clas
         // }
         else if (typeof type === 'object') {
             if (Enum.isEnum(type)) {
-                (attrs || initAttrs())[`${propertyNamePrefix}type`] = ENUM_TAG;
-                attrs![`${propertyNamePrefix}enumList`] = Enum.getList(type);
+                setPropertyEnumTypeOnAttrs(
+                    attrs || initAttrs(),
+                    propertyName,
+                    type,
+                );
```

Comment:
I don't think this is a logic that specifically belongs to the attr module. Because it was always done on CCClass. Like the code for registering BitMask. So I think setPropertyEnumTypeOnAttrs this should be internally implementation and use in CCClass. Then expose the setPropertyEnumType on CCClass.
This also prevents external functions from accessing '... /core/internal-index', which is such a weird module.

Reply:
Putting `setPropertyEnumType` onto `CCClass` does expose it as public, is it your intention?

Reply:
Yes

Reply:
I think it is better export as less API as possible if it is not needed.